tests/unit/processor/requester/test_requester.py

```diff
@@ -0,0 +1,274 @@
+        {"message": "the message", "tags": ["_requester_failure"]},
+        {
+            "method": "GET",
+            "url": "http://failure_mock",
+            "body": HTTPError("404"),
+            "content_type": "text/plain",
+            "status": 404,
+        },
+    ),
+    (
+        "timout error",
+        {
+            "filter": "message",
+            "requester": {"url": "http://failure_mock", "method": "GET", "timeout": 0.2},
+        },
+        {"message": "the message"},
+        {"message": "the message", "tags": ["_requester_failure"]},
+        {
+            "method": "GET",
+            "url": "http://failure_mock",
+            "body": ConnectTimeout(),
+            "content_type": "text/plain",
+            "status": 200,
+        },
+    ),
+    (
+        "does not overwrite if not permitted",
+        {
+            "filter": "message",
+            "requester": {"url": "http://failure_mock", "method": "GET", "target_field": "result"},
+        },
+        {"message": "the message", "result": "preexisting"},
+        {"message": "the message", "result": "preexisting", "tags": ["_requester_failure"]},
+        {
+            "method": "GET",
+            "url": "http://failure_mock",
+            "body": "the body",
+            "content_type": "text/plain",
+            "status": 200,
+        },
+    ),
+    (
+        "errors on missing fields",
+        {
+            "filter": "message",
+            "requester": {"url": "http://${missingfield}", "method": "GET"},
+        },
+        {"message": "the message"},
+        {"message": "the message", "tags": ["_requester_failure"]},
+        {},
+    ),
+]  # testcase, rule, event, expected, mock
+
+
+class TestRequester(BaseProcessorTestCase):
+
+    CONFIG: dict = {
+        "type": "requester",
+        "specific_rules": ["tests/testdata/unit/requester/specific_rules"],
+        "generic_rules": ["tests/testdata/unit/requester/generic_rules"],
+    }
+
+    @responses.activate
+    @pytest.mark.parametrize("testcase, rule, event, expected, response_kwargs", test_cases)
+    def test_testcases(
+        self, testcase, rule, event, expected, response_kwargs
+    ):  # pylint: disable=unused-argument
+        responses.add(responses.Response(**response_kwargs))
+        self._load_specific_rule(rule)
+        self.object.process(event)
+        assert event == expected
+
+    @responses.activate
+    @pytest.mark.parametrize("testcase, rule, event, expected, response_kwargs", failure_test_cases)
+    def test_requester_testcases_failure_handling(
+        self, testcase, rule, event, expected, response_kwargs
+    ):
+        if response_kwargs:
+            responses.add(responses.Response(**response_kwargs))
+        self._load_specific_rule(rule)
+        with pytest.raises(ProcessingWarning):
+            self.object.process(event)
+        assert event == expected, testcase
+
+    @responses.activate
+    def test_process(self):
```

Comment:
What is this test case for? There is only one assert that should be always true and after the other calls there is nothing checked again.

Reply:
I had to override this from the base class, to decorate it with the responses fixture